core/src/main/java/org/kiwix/kiwixmobile/core/bookmark/BookmarksActivity.java

```diff
@@ -158,30 +172,31 @@ protected void onResume() {
 
   @Override
   public boolean onCreateOptionsMenu(Menu menu) {
-    getMenuInflater().inflate(R.menu.menu_bookmarks, menu);
-    MenuItem toggle = menu.findItem(R.id.menu_bookmarks_toggle);
-    toggle.setChecked(sharedPreferenceUtil.getShowBookmarksCurrentBook());
-
-    SearchView search = (SearchView) menu.findItem(R.id.menu_bookmarks_search).getActionView();
-    search.setQueryHint(getString(R.string.search_bookmarks));
-    search.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
-      @Override
-      public boolean onQueryTextSubmit(String query) {
-        return false;
-      }
 
-      @Override
-      public boolean onQueryTextChange(String newText) {
-        bookmarksList.clear();
-        bookmarksList.addAll(allBookmarks);
-        if ("".equals(newText)) {
-          bookmarksAdapter.notifyDataSetChanged();
+    if (!bookmarksList.isEmpty()) {
```

Comment:
This has introduced a race condition. All the bookmarks need to be read from the database before the menu is created or else we will never see it.

In BookMarksPresenter I added this delay:
```
 @Override
  public void loadBookmarks(boolean showFromCurrentBookmarks) {
    dataSource.getBookmarks(showFromCurrentBookmarks).delay(1, TimeUnit.SECONDS)
```
and nothing was ever displayed. 

Please update HistoryActivity as well

Reply:
@macgills I didn't find this line present in the files currently. And on adding it nothing is displayed.
 
I am not clear as to how a race condition is created. All bookmarks need not be read before the menu is created. Even if one is read, it is enough to show that the bookmarksList is not empty.

Reply:
>I added this delay
I checked out your branch to test what would happen if the database read took 1 second.

OnCreateOptionsMenu is not called by us, it is an android lifecycle method. I think it happens after onStart but I am not 100% when it happens with respect to the common activity lifecycle methods. OnCreateOptionsMenu can be invoked again by `invalidateOptionsMenu` and `onPrepareOptionsMenu` exists to handle some rendering cases. 
I prefer to create once, hold onto references to the menuItems and change their properties as necessary. This can be irritating as they are potentially null.

For your code to work the database read must complete before onCreateOptionsMenu happens

Reply:
Deeply odd behaviour, the list won't appear until you try to scroll it. This entire screen should really be overhauled, far too complex for what it is but that is not important right now.

![image](https://user-images.githubusercontent.com/3358282/73657544-63156880-468a-11ea-844f-e0d8cea916fe.png)
The important thing is that I can make the list appear but you can't search or delete and that is a result of changes in this PR.

Reply:
@macgills for now, I will remove the condition of not showing the menu if the bookmarks list is empty then? That shouldn't be a problem I think. Later, I'll try to find out why this behavior is happening and fix it.

Reply:
Sounds good to me

Reply:
@macgills done.